internal/k8s/client.go

```diff
@@ -122,3 +123,15 @@ func (k KubectlClient) Delete(ctx context.Context, entities []K8sEntity) error {
 	}
 	return err
 }
+
+func (k KubectlClient) applyOrDeleteFromEntities(ctx context.Context, cmd string, entities []K8sEntity) (stdout string, stderr string, err error) {
+	args := []string{cmd, "-f", "-"}
+
+	rawYAML, err := SerializeYAML(entities)
+	if err != nil {
+		return "", "", fmt.Errorf("serializeYaml for kubectl %s: %v", cmd, err)
+	}
+	stdin := bytes.NewReader([]byte(rawYAML))
```

Comment:
it might be a bit cleaner to change `execWithStdin` to take `stdin` as a `string` directly (or if we're concerned about not having all the input at once, an `io.Reader`)